### PR TITLE
Windows setup with errors so far

### PR DESCRIPTION
**Pull request by frigoref**
Fixes for solved some issues, but not all yet ...
- Current Google Java Format Plugin version works
- gradlew.bat calls needed

### CODE REVIEW

File: docs/development/how-to/ide-setup/eclipse.md [hunk 1]
```diff
@@ -53,5 +53,22 @@ Should look like this when configured: <br />
   - [EGit in Eclipse](http://www.eclipse.org/egit/) - with [tutorial
       ](http://www.vogella.com/tutorials/EclipseGit/article.html)
   - Run Configurations are checked in
-  - Run `./gradlew downloadAssets` one time to get images downloaded
-
+  - Run `./gradlew downloadAssets` (Windows: `gradlew.bat downloadAssets`) once in the main triplea folder to get images downloaded
+  - Run also the following commands (Windows: `gradlew.bat`) once in the main triplea folder if errors occur during gradle build for any of the tasks
+    - `./gradlew :spotlessApply`
```
Comment (DanVanAtta) [inline docs/development/how-to/ide-setup/eclipse.md:58]: `./gradlew :spotlessApply` should recursively apply to all sub-projects. Did that not work? The enumeration of sub-projects should be unnecessary...

File: docs/development/how-to/ide-setup/eclipse.md [hunk 2]
```diff
@@ -53,5 +53,22 @@ Should look like this when configured: <br />
   - [EGit in Eclipse](http://www.eclipse.org/egit/) - with [tutorial
       ](http://www.vogella.com/tutorials/EclipseGit/article.html)
   - Run Configurations are checked in
-  - Run `./gradlew downloadAssets` one time to get images downloaded
-
+  - Run `./gradlew downloadAssets` (Windows: `gradlew.bat downloadAssets`) once in the main triplea folder to get images downloaded
+  - Run also the following commands (Windows: `gradlew.bat`) once in the main triplea folder if errors occur during gradle build for any of the tasks
```
Comment (DanVanAtta) [inline docs/development/how-to/ide-setup/eclipse.md:57]: > "if errors occur during gradle build for any of the tasks"

Which errors were you seeing?

File: docs/development/how-to/ide-setup/eclipse.md [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
 ## Formatter
 1. [Install Google Java Format Plugin](https://github.com/google/google-java-format#eclipse)
-   * ***Important*** Use the unofficial 1.7 version available here: https://github.com/google/google-java-format/files/2774507/google-java-format-eclipse-plugin-1.7.0.jar.zip
+   *[OLD ***Important*** Use the unofficial 1.7 version available here: https://github.com/google/google-java-format/files/2774507/google-java-format-eclipse-plugin-1.7.0.jar.zip]
```
Comment (DanVanAtta) [inline docs/development/how-to/ide-setup/eclipse.md:26]: I'm curious why is this old - is there a new plugin now that makes this old? Were you able to install the formatter? 

Side-note: i'm very tempted to tell everyone to always use Intellij IDE, it is also free and it's easier to configure. Intellij is also rapidly gaining market share, eclipse is somewhat, 'eclipsing'. A similar call out for the windows commands are probably needed on the intellij setup page as well.

Comment (DanVanAtta) [reply docs/development/how-to/ide-setup/eclipse.md:26]: Ah, I see: 

> Current Google Java Format Plugin version works

Then please delete this line. Having 'old' documentation raises a lot more questions than it answers : +)

File: docs/development/how-to/ide-setup/eclipse.md [hunk 4]
```diff
@@ -53,5 +52,6 @@ Should look like this when configured: <br />
   - [EGit in Eclipse](http://www.eclipse.org/egit/) - with [tutorial
       ](http://www.vogella.com/tutorials/EclipseGit/article.html)
   - Run Configurations are checked in
-  - Run `./gradlew downloadAssets` one time to get images downloaded
-
+  - Run `./gradlew downloadAssets` (Windows: `gradlew.bat downloadAssets`) once in the main triplea folder to get images downloaded
+  - Run `./gradlew downloadAssets` (Windows: `gradlew.bat downloadAssets`) once in the main triplea folder to get images downloaded
```
Comment (DanVanAtta) [inline docs/development/how-to/ide-setup/eclipse.md:56]: Looks like this line got duplicated

File: docs/development/how-to/ide-setup/eclipse.md [hunk 5]
```diff
@@ -53,5 +52,6 @@ Should look like this when configured: <br />
...
+  - Run `./gradlew spotlessApply` (Windows: `gradlew.bat spotlessApply`) at least once in the main triplea folder if errors occur during gradle build for any of the tasks (it might have to be run multiple times though)
```
Comment (DanVanAtta) [inline docs/development/how-to/ide-setup/eclipse.md:57]: I suspect this was due to running on windows, needing to run spotlessApply arguably is not related to the eclipse setup. 

Without knowing the exact errors you were hitting @frigoref , I suspect they were line ending problems. Those can be solved via a local config: https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings and would not require you to repeatedly run spotlessApply to fix.

File: docs/development/how-to/ide-setup/eclipse.md [hunk 6]
```diff
@@ -23,8 +23,7 @@
 
 ## Formatter
 1. [Install Google Java Format Plugin](https://github.com/google/google-java-format#eclipse)
-   * ***Important*** Use the unofficial 1.7 version available here: https://github.com/google/google-java-format/files/2774507/google-java-format-eclipse-plugin-1.7.0.jar.zip
-   * Download the 1.7 version, unzip it, and move the extracted 'jar' file to the eclipse 'dropins' folder.
+   * Download the most recent version, unzip it, and move the extracted 'jar' file to the eclipse 'dropins' folder.
```
Comment (DanVanAtta) [inline docs/development/how-to/ide-setup/eclipse.md:26]: Is this comment still necessary? It looks like anyone following the link would read the same information: https://github.com/google/google-java-format#eclipse

FWIW, to explain my thinking here: for documentation, I certainly believe 'less is more'. More documentation is more things to maintain, and more stuff for a new person to sift through to find the really useful information (and so lots of documentation lowers the value of the useful documentation, makes it harder to find).

Comment (frigoref) [reply docs/development/how-to/ide-setup/eclipse.md:26]: In general I agree that 'less is more', but having multiple links to open and keeping an overview is making it more cumbersome. It is more user-friendly (especially for new users) to have one complete guide on a page that they can follow. 
Even if this guide will be incorrect over time, the user still can make/request updates (same as for broken links) or can try to figure it out for himself with the link provided.

File: docs/development/how-to/ide-setup/eclipse.md (hunk 4 above)
Comment (frigoref) [reply docs/development/how-to/ide-setup/eclipse.md]: Fixed.

File: docs/development/how-to/ide-setup/eclipse.md (hunk 5 above)
Comment (frigoref) [reply docs/development/how-to/ide-setup/eclipse.md:56]: I do not understand what you mean, but I gut error messages saying "_Run gradlew :xxx:spotlessApply_" several times. First I ran them individually and then I just used _gradlew spotlessApply_

Comment (DanVanAtta) [reply docs/development/how-to/ide-setup/eclipse.md:56]: - Needing to run 'spotlessApply (formatting) is not relevant to IDE setup. The formatting is fine after checkout.

- An exact error message would be helpful. I suspect the problem are line endings being silently converted when you save files. That is an OS and configuration problem and can be fixed.

File: docs/development/how-to/ide-setup/eclipse.md (hunk 6 above)
Comment (DanVanAtta) [reply docs/development/how-to/ide-setup/eclipse.md:26]: > but having multiple links to open and keeping an overview is making it more cumbersome.

How do you mean by this? Which multiple links exactly?

**Comment by frigoref** [comment]
Done in other pull request.

**Pull request closed without merging**